File: IT/Database/Esercizi/Python/esercizio13.py
# file esercizio13.py

from __future__ import print_function
import sqlite3

conn = sqlite3.connect("students.db")
cursor = conn.cursor()

try:
	cursor.execute("SELECT AVG(grade) FROM students")
	v = cursor.fetchone()[0]

	if not v:
		print("Nessuno studente ha ancora passato l'esame.")
		exit()

	print("Voto medio: {}".format(v))

except sqlite3.OperationalError as e:
	print(e)

File: IT/2/Database/Esercizi/C++/esercizio13.cpp
/* file esercizio13.cpp */

#include <iostream>
#include <sqlite3.h>

using std::cout;
using std::endl;

int main()
{
	sqlite3 *db;
	sqlite3_stmt *res;

	sqlite3_open("students.db", &db);
	int rc = sqlite3_prepare_v2(db, "SELECT AVG(grade) FROM students", -1, &res, NULL);

	int avg = 0;

	if (rc != SQLITE_ERROR) {
		if (sqlite3_step(res) == SQLITE_ROW) {
			avg = sqlite3_column_double(res, 0);
		}

		if (!avg) {
			cout << "Nessuno studente ha ancora passato l'esame." << endl;
			sqlite3_finalize(res);
			sqlite3_close(db);
			return 0;
		}

		cout << "Voto medio: " << avg << endl;
	}
	else {
		cout << sqlite3_errmsg(db) << endl;
	}

	sqlite3_finalize(res);
	sqlite3_close(db);
}

